pkg/apis/kops/validation/aws.go

```diff
@@ -241,6 +241,16 @@ func awsValidateLoadBalancerSubnets(fieldPath *field.Path, spec kops.ClusterSpec
 				allErrs = append(allErrs, field.Forbidden(fieldPath.Index(i).Child("privateIPv4Address"), "privateIPv4Address only allowed for internal NLBs"))
 			}
 		}
+
+		if subnet.AllocationID != nil {
+			if *subnet.AllocationID == "" {
```

Comment:
Can we use fi.StringValue over here? Something like:
```
if fi.StringValue(subnet.AllocationID) == "" {

}
```
That takes care of the case when subnet.AllocationID is nil too?

Reply:
My GO isn't that great so I'm not sure how much of a difference there is between `!= nil` and ` ==""`

This conditional was mostly a copy of the above `PrivateIPv4Address` but without the `ParseIP` and switching the condition on the `lbSpec.Type` to require a Public rather than Internal NLB

I'm happy to switch to `if fi.StringValue(subnet.AllocationID) != "" {` and keep the `lbSpec` conditionals on 250.
Does `!= ""` cover where AllocationID is `nil` or undefined value

Reply:
Yup I ll paste the implementation of fi.StringValue here:
```
func StringValue(s *string) string {
	if s == nil {
		return ""
	}
	return *s
}
```
subnet.AllocationId!=nil checks if the string pointer is nil but subnet.AllocationId=="" checks if the pointer points to an empty string. The string pointer can be non nil but the string which the pointer is pointing to can be "". fi.StringValue returns "" if the pointer nil. So it seems cleaner. You can avoid 2 if stmts and make it to one.....

Reply:
I think fi.StringValue should take care of nil case too. I think its nicer since almost the entire kOps codebase uses it.... Do try it out once...

Reply:
@bharath-123 Please take a look. 
I've reduced the conditional. As `fi.StringValue(subnet.AllocationID) != ""` is also covering the set but empty string I only needed to handle the loadbalancer being of the wrong type (classic or internal)

Reply:
Got it. I missed this use case of validating AllocationId to not be empty when it is present.